### ISSUE TITLE
Factor levels are not recoverable upon prediction when a stack only contains the Lrnr_mean

### ISSUE THREAD

**Issue by carpenitoThomas** [tlverse/sl3#303]
When you create a stack that **only** includes the `Lrnr_mean` function and you are predicting a categorical outcome, the returned predictions do not include the names of the factor levels. Here is a reproducible example where you can see that levels from the predictions are lost:
```{r}
library(tidyverse)

data(cpp)

cpp <- cpp %>%
  select(c(bmi, agedays, feeding)) %>%
  mutate(feeding = as.factor(feeding)) %>%
  na.omit

task <- make_sl3_Task(cpp, covariates = c("agedays", "bmi"), outcome = "feeding")

Lrnr_mean <- Lrnr_mean$new()
stack <- make_learner(Stack, Lrnr_mean)

sl <- sl3::Lrnr_sl$new(learners = stack)
stack_fit <- sl$train(task)

predictions <- stack_fit$predict()
names(predictions[[1]][[1]])
```
Note, when you include another learner in the stack the factor levels are retained:

```{r}
Lrnr_glmnet <- Lrnr_glmnet$new()
stack2 <- make_learner(Stack, Lrnr_mean, Lrnr_glmnet)
sl2 <- sl3::Lrnr_sl$new(learners = stack2)
stack_fit2 <- sl2$train(task)

predictions2 <- stack_fit2$predict()
names(predictions2[[1]][[1]])
```

** edited to correct code example.

**Comment by carpenitoThomas** [comment]
I do not believe this is fail on Travis is due to this code? I downloaded a fresh copy of the repository and it fails on my machine as well. If this is not the case please let me know.

**Comment by nhejazi** [comment]
Resolved by #306. Thanks for reporting and fixing!
